### DISCUSSION TITLE
Doobie SQL injection query

### DISCUSSION THREAD

**Discussion by domaspoliakas** [tree-sitter/tree-sitter-scala#315] [Q&A]
Hi folks! 👋 

Prompted by a question on the typelevel discord I added these doobie SQL injection queries into my local neovim config

```lisp
; extends

(interpolated_string_expression 
  interpolator: (identifier) @_id (#eq? @_id "sql")
  (interpolated_string) @sql
  )

(interpolated_string_expression 
  interpolator: (identifier) @_id (#eq? @_id "fr")
  (interpolated_string) @sql
  )
```

I was wondering whether something like this should and would get merged into the injections on this repo. There are many doobie users who could benefit from something like this. Having said that, this being library-specific I can understand if there are reservations. And if the consensus would be not to merge - is there somewhere in this repo where things like this could be documented for future users?

(these are ofc nvim specific queries, but more idiomatic queries can be easily made out of these too)

Thanks for your time!

**Comment by susliko** [comment]
Curious
Personally I have no objections for such library-specific support (e.g. we already have support for scala-cli comments), but I tried it and highlighting looked too motley to me. I'd prefer this not being default
<img width="418" alt="image" src="https://github.com/tree-sitter/tree-sitter-scala/assets/15569000/698a96a1-46ac-43ca-8521-834137267ac7">

**Reply by domaspoliakas** [reply]
Yeah, after using it for a bit - it definitely has rough edges due to the simple fact that it's not _really_ SQL - e.g. quotes have to be escaped, and the SQL grammar does not like that. 

Having said that I sanded off a couple of edges:

```lisp
(interpolated_string_expression 
  interpolator: (identifier) @_id (#any-of? @_id "sql" "fr" "fr0")
  (interpolated_string) @sql (#offset! @sql 0 3 0 -1) (#match? @sql "^\"\"\".*")
  )

(interpolated_string_expression 
  interpolator: (identifier) @_id (#any-of? @_id "sql" "fr" "fr0")
  (interpolated_string) @sql (#offset! @sql 0 1 0 -1) (#match? @sql "^\"\[^\"].*")
  )
```

Maybe this will work better? But it definitely is less tree-sitter and more nvim-treesitter at that point

**Reply by susliko** [reply]
Looks like this discussion is somewhat similar to the one in https://github.com/tree-sitter/tree-sitter-scala/pull/286 about system-specific queries
